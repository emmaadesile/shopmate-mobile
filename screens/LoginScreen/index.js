import React from "react";
import { TouchableOpacity, View } from "react-native";

import {
  Container,
  Logo,
  BodyText,
  Form,
  Input,
  LoginButton,
  SubText,
  ButtonText
} from "./styles";

class LoginScreen extends React.Component {
  static navigationOptions = {
    header: null
  };

  render() {
    return (
      <Container>
        <Logo source={require("../../assets/storex_logo.png")} />
        <BodyText>Login</BodyText>
        <Form>
          <Input placeholder="Email" keyboardType="email-address" />
          <Input placeholder="Password" secureTextEntry={true} />
          <TouchableOpacity
            style={{
              width: "100%"
            }}
          >
            <LoginButton>
              <ButtonText>Sign in</ButtonText>
            </LoginButton>
          </TouchableOpacity>
          <View
            style={{
              justifyContent: "center",
              alignItems: "center",
              flexDirection: "row"
            }}
          >
            <SubText>Not a member?</SubText>
            <TouchableOpacity onPress={() => this.props.navigation.goBack()}>
              <SubText color="#EFB961">Sign up</SubText>
            </TouchableOpacity>
          </View>
        </Form>
      </Container>
    );
  }
}

export default LoginScreen;
