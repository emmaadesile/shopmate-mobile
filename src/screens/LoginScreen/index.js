import React from "react";
import { TouchableOpacity, View } from "react-native";
import { validateLoginForm } from "../../helpers/formValidation";

import {
  Container,
  Logo,
  BodyText,
  Form,
  Input,
  LoginButton,
  SubText,
  ButtonText
} from "./styles";

class LoginScreen extends React.Component {
  static navigationOptions = {
    header: null
  };

  state = {
    email: "",
    password: "",
    isEmailFocused: false,
    isPasswordFocused: false,
    isValid: false,
    error: {}
  };

  onEmailFocusChange = () => {
    this.setState(prevState => ({ isEmailFocused: !prevState.isEmailFocused }));
  };

  onPasswordFocusChange = () => {
    this.setState(prevState => ({
      isPasswordFocused: !prevState.isPasswordFocused
    }));
  };

  validateForm = () => {
    const { email, password } = this.state;
    const { isValid, error } = validateLoginForm({ email, password });

    this.setState(prevState => ({
      ...prevState,
      error,
      isValid
    }));
  };

  render() {
    return (
      <Container>
        <Logo source={require("../../assets/storex_logo.png")} />
        <BodyText>Sign in to your Account</BodyText>
        <Form>
          <Input
            onFocus={this.onEmailFocusChange}
            onBlur={this.onEmailFocusChange}
            onChangeText={email => {
              this.setState({ email });
              this.validateForm();
            }}
            value={this.state.email}
            placeholder="Email"
            keyboardType="email-address"
            borderColor={this.state.isEmailFocused ? "#efb961" : "#d8d8d8"}
          />
          <Input
            onFocus={this.onPasswordFocusChange}
            onBlur={this.onPasswordFocusChange}
            onChangeText={password => {
              this.setState({ password });
              this.validateForm();
            }}
            value={this.state.password}
            placeholder="Password"
            secureTextEntry={true}
            borderColor={this.state.isPasswordFocused ? "#efb961" : "#d8d8d8"}
          />
          <TouchableOpacity
            disabled={!this.state.isValid}
            style={{
              width: "100%"
            }}
          >
            <LoginButton
              backgroundColor={!this.state.isValid && "rgba(239,185,97,0.44)"}
            >
              <ButtonText> Sign in</ButtonText>
            </LoginButton>
          </TouchableOpacity>
          <View
            style={{
              justifyContent: "center",
              alignItems: "center",
              flexDirection: "row"
            }}
          >
            <SubText>Not a member?</SubText>
            <TouchableOpacity onPress={() => this.props.navigation.goBack()}>
              <SubText color="#EFB961"> Sign up</SubText>
            </TouchableOpacity>
          </View>
        </Form>
      </Container>
    );
  }
}

export default LoginScreen;
