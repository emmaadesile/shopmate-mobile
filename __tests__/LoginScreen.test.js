import React from "react";
import renderer from "react-test-renderer";
import LoginScreen from "../src/screens/LoginScreen";

describe.skip("Login Screen", () => {
  it("renders the the login screen", () => {
    const component = renderer.create(<LoginScreen />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
