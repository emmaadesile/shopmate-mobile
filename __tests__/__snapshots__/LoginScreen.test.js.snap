// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Screen renders the the login screen 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "center",
        "paddingLeft": 20,
        "paddingRight": 20,
        "width": "100%",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 54,
          "width": 114,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#efb961",
          "fontSize": 24,
          "fontWeight": "normal",
          "height": 62,
          "marginTop": 40,
          "textAlign": "center",
          "width": 300,
        },
      ]
    }
  >
    Sign in to your Account
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      borderColor="#d8d8d8"
      keyboardType="email-address"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "borderColor": "#d8d8d8",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 15,
            "height": 50,
            "marginBottom": 15,
            "paddingLeft": 15,
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      borderColor="#d8d8d8"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "borderColor": "#d8d8d8",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 15,
            "height": 50,
            "marginBottom": 15,
            "paddingLeft": 15,
            "width": "100%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        backgroundColor="rgba(239,185,97,0.44)"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(239,185,97,0.44)",
              "borderRadius": 8,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 10,
              "shadowColor": "rgba(0, 0, 0, 0.15)",
              "shadowOffset": Object {
                "height": 10,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 20,
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
                "letterSpacing": 2,
                "textTransform": "uppercase",
              },
            ]
          }
        >
           Sign in
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#c4c4c4",
              "fontSize": 17,
              "marginTop": 25,
            },
          ]
        }
      >
        Not a member?
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          color="#EFB961"
          style={
            Array [
              Object {
                "color": "#EFB961",
                "fontSize": 17,
                "marginTop": 25,
              },
            ]
          }
        >
           Sign up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
